src/cpp/r/R/Tools.R

```diff
@@ -286,61 +286,66 @@ assign(envir = .rs.Env, ".rs.hasVar", function(name)
 # restore an object from a file
 .rs.addFunction( "restoreGraphics", function(filename)
 {
-   load(filename)
+   # load the 'plot' object
+   envir <- new.env(parent = emptyenv())
+   load(filename, envir = envir)
+   plot <- envir$plot
    
-   # restore native symbols for R >= 3.0
+   # restore native symbols
+   dlls <- getLoadedDLLs()
    rVersion <- getRversion()
-   if (rVersion >= "3.0")
-   {
-      for(i in 1:length(plot[[1]]))
-      {
-         # get the symbol then test if it's a native symbol
-         symbol <- plot[[1]][[i]][[2]][[1]]
-         if("NativeSymbolInfo" %in% class(symbol))
-         {
-            # determine the dll that the symbol lives in
-            if (!is.null(symbol$package))
-               name = symbol$package[["name"]]
-            else
-               name = symbol$dll[["name"]]
-            pkgDLL <- getLoadedDLLs()[[name]]
-
-            # reconstruct the native symbol and assign it into the plot
-            nativeSymbol <-getNativeSymbolInfo(name = symbol$name,
-                                               PACKAGE = pkgDLL,
-                                               withRegistrationInfo = TRUE);
-            plot[[1]][[i]][[2]][[1]] <- nativeSymbol;
-         }
-      }
-   }
-   # restore native symbols for R >= 2.14
-   else if (rVersion >= "2.14")
-   {
-     try({
-       for(i in 1:length(plot[[1]])) 
-       {
-         if("NativeSymbolInfo" %in% class(plot[[1]][[i]][[2]][[1]]))
-         {
-           nativeSymbol <-getNativeSymbolInfo(plot[[1]][[i]][[2]][[1]]$name);
-           plot[[1]][[i]][[2]][[1]] <- nativeSymbol;         
-         }
-       }
-     },
-     silent = TRUE);
-   }
-
-   # set the pid attribute to the current pid if necessary
-   if (rVersion >= "3.0.2")
```

Comment:
Super nit: we use "3.0.1" as the required version in a number of places. Presumably this requires 3.0.2. Do you think we should make that tiny tweak, or do we have a sufficient case for requiring 3.1 starting with 1.3? (the symmetry alone is pleasing...)

Reply:
I _think_ this change is probably fine, either way -- I'm suspecting older versions of R would just ignore the attribute. But I could be wrong...

I think we're probably okay leaving the minimum version at R 3.0.1 for now, though. At least, this particular PR isn't enough to force the version up, IMHO.

Reply:
Okay, sounds good!